corehq/apps/userreports/models.py

```diff
@@ -683,6 +687,48 @@ def _get_report_config(cls, static_config, domain):
         return ReportConfiguration.wrap(doc)
 
 
+class AsyncIndicator(models.Model):
+    doc_id = models.CharField(max_length=255, null=False, db_index=True)
+    doc_type = models.CharField(max_length=126, null=False)
+    domain = models.CharField(max_length=126, null=False)
+    pillow = models.CharField(max_length=126, null=False)
+    indicator_config_ids = ArrayField(
+        models.CharField(max_length=126, null=True, blank=True),
+        null=False
+    )
+    date_created = models.DateTimeField(auto_now_add=True, db_index=True)
+
+    class Meta(object):
+        unique_together = ('doc_id', 'pillow',)
+        ordering = ["date_created"]
+
+    @classmethod
+    def update_indicators(cls, change, pillow, table_ids):
+        doc_id = change.id
+        pillow_id = pillow.pillow_id
+        with CriticalSection([get_async_indicator_modify_lock_key(doc_id, pillow_id)]):
```

Comment:
I don't think so. Locks are want we would want in the following situation:

add indicator for doc_id 1, pillow static, table_id first_table
indicator begins processing in save_document
save_document begins processing
update this indicator to add table_id second_table (the record that save_document has only grabbed that table)
save_document ends processing
save_document deletes the record without processing second_table

I'm not sure of a good way to solve that without locks because UCRs can be over mulitple sql databases or ES so there wouldn't be a way to roll those back without a large refactor